roles/certificates/tasks/main.yml

```diff
@@ -1,132 +1,7 @@
-
-- name: 'Creating CA certificate and key'
-  ansible.builtin.command: >
-    openssl req -new
-      -x509
-      -extensions v3_ca
-      -config "{{ certificates_ca_directory }}/openssl.cnf"
-      -keyout "{{ certificates_ca_directory_keys }}/ca.key"
-      -out "{{ certificates_ca_directory_certs }}/ca.crt"
-      -passout "file:{{ certificates_ca_directory_keys }}/ca.pwd"
-  args:
-    creates: "{{ certificates_ca_directory_certs }}/ca.crt"
-
-- name: 'Create server key'
-  ansible.builtin.command: >
-    openssl genrsa
-      -out "{{ certificates_ca_directory_keys }}/{{ certificates_server }}.key"
-  args:
-    creates: "{{ certificates_ca_directory_keys }}/{{ certificates_server }}.key"
-
-- name: 'Create server signing request'
-  ansible.builtin.command: >
-    openssl req
-      -new
-      -config "{{ certificates_ca_directory }}/openssl.cnf"
-      -key "{{ certificates_ca_directory_keys }}/{{ certificates_server }}.key"
-      -out "{{ certificates_ca_directory_requests }}/{{ certificates_server }}.csr"
-  args:
-    creates: "{{ certificates_ca_directory_requests }}/{{ certificates_server }}.csr"
-
-- name: 'Create server certificate'
-  ansible.builtin.command: >
-    openssl ca
-      -create_serial
-      -batch
-      -extensions ssl_server
-      -passin "file:{{ certificates_ca_directory_keys }}/ca.pwd"
-      -config "{{ certificates_ca_directory }}/openssl.cnf"
-      -in "{{ certificates_ca_directory_requests }}/{{ certificates_server }}.csr"
-      -out "{{ certificates_ca_directory_certs }}/{{ certificates_server }}.crt"
-  args:
-    creates: "{{ certificates_ca_directory_certs }}/{{ certificates_server }}.crt"
-
-- name: 'Create client key'
-  ansible.builtin.command: >
-    openssl genrsa
-      -out "{{ certificates_ca_directory_keys }}/{{ certificates_client }}.key"
-  args:
-    creates: "{{ certificates_ca_directory_keys }}/{{ certificates_client }}.key"
-
-- name: 'Create client signing request'
-  ansible.builtin.command: >
-    openssl req
-      -new
-      -config "{{ certificates_ca_directory }}/openssl.cnf"
-      -key "{{ certificates_ca_directory_keys }}/{{ certificates_client }}.key"
-      -out "{{ certificates_ca_directory_requests }}/{{ certificates_client }}.csr"
-  args:
-    creates: "{{ certificates_ca_directory_requests }}/{{ certificates_client }}.csr"
-
-- name: 'Create server certificate'
-  ansible.builtin.command: >
-    openssl ca
-      -create_serial
-      -batch
-      -extensions ssl_client
-      -passin "file:{{ certificates_ca_directory_keys }}/ca.pwd"
-      -config "{{ certificates_ca_directory }}/openssl.cnf"
-      -in "{{ certificates_ca_directory_requests }}/{{ certificates_client }}.csr"
-      -out "{{ certificates_ca_directory_certs }}/{{ certificates_client }}.crt"
-  args:
-    creates: "{{ certificates_ca_directory_certs }}/{{ certificates_client }}.crt"
+- name: 'Generate CA certificate'
+  ansible.builtin.include_tasks: ca.yml
+  when: certificates_ca
```

Comment:
Under which circumstances would this be false?

Reply:
When generating a certificate bundle for use on proxies - https://github.com/theforeman/foreman-quadlet/pull/90/files#diff-b1491adbed08c8b00b570245e695c525e8dbf2a9ef94181d292272ffae52b5a5R6

Reply:
Ah, so you ran this once with `certificates_ca: true` to create a CA, and then you run it with `false` (on the same machine) to create the bundle (with a new hostname -- for a different machine) without regenerating the CA?

Does that imply ca.yml is not idempotent? Or is that pure speed optimization?

Reply:
Speed and safety optimization I suppose. I was also not sure how the design would fall out where something else provided the certificates yet (e.g. puppet-certs). I am still not 100% and this will get re-factored as we iterate.